yetanotherco/aligned_layer
# feat: add batcher queue len and queue size in bytes metrics

# Add batcher queue len and queue size in bytes metrics

## Description

This PR adds two new metrics for the batcher:
* Queue length: The number of proofs waiting in the batcher's queue to be included in a batch
* Queue size: The sum of the sizes of the verification data, measured in bytes, of the proofs waiting in the batcher's queue.

<img width="760" alt="image" src="https://github.com/user-attachments/assets/c074798d-d52f-40ae-820f-70ea0cb7e8af">

## How to test

1. Start all services locally, including the metrics
2. Start sending proofs
3. Go to the "System Data" dashboard and go to the bottom on the left size (the batcher side). You should see four new graphs showing the queue len and queue size.
4. Stop sending proofs and the metrics should be reset back to zero.
5. Send a burst of many proofs to see a spike in the graphs with `make batcher_send_sp1_burst BURST_SIZE=23`

## Type of change

Please delete options that are not relevant.


## Checklist

  - [Link to the open issue addressing it]()
